src/models/dashboard.js

```diff
@@ -189,26 +189,30 @@ export default {
 
       response.forEach(timeseriesResponse => {
         const timeseriesCollection = [];
-        const firstResponse = timeseriesResponse.hits.hits[0]._source;
-        const runId = firstResponse.run.id;
-        const primaryMetric = firstResponse.sample.measurement_title;
-        const iterationName = firstResponse.iteration.name;
-        const sampleName = firstResponse.sample.name;
-        timeseriesResponse.hits.hits.forEach(timeseries => {
-          timeseriesCollection.push({
-            x: timeseries._source['@timestamp_original'],
-            [`y-${runId}_${iterationName}_${sampleName}`]: timeseries._source.result.value,
+        // gets firstResponse only when there is a
+        // non empty array of hits.
+        if (timeseriesResponse.hits.hits.length > 0) {
+          const firstResponse = timeseriesResponse.hits.hits[0]._source;
+          const runId = firstResponse.run.id;
+          const primaryMetric = firstResponse.sample.measurement_title;
+          const iterationName = firstResponse.iteration.name;
+          const sampleName = firstResponse.sample.name;
+          timeseriesResponse.hits.hits.forEach(timeseries => {
+            timeseriesCollection.push({
+              x: timeseries._source['@timestamp_original'],
+              [`y-${runId}_${iterationName}_${sampleName}`]: timeseries._source.result.value,
+            });
           });
-        });
 
-        Object.entries(clusteredIterations[primaryMetric]).forEach(([clusterId, cluster]) => {
-          const clusterKey = `${runId}_${iterationName}_${sampleName}`;
-          if (clusterKey in cluster) {
-            clusteredIterations[primaryMetric][clusterId][
-              clusterKey
-            ].timeseries = timeseriesCollection;
-          }
-        });
+          Object.entries(clusteredIterations[primaryMetric]).forEach(([clusterId, cluster]) => {
+            const clusterKey = `${runId}_${iterationName}_${sampleName}`;
```

Comment:
@dbutenhof I'm sorry, I meant the request may be wrong, the request to the ES ep is ```request.post(``${endpoint}&scroll_id=${allData._scroll_id}``)```, which returns a 502

Reply:
Sorry; I didn't mean to suggest this comment has anything to do with your problem.

Reply:
I'll try again: my point was that it seems that line 208 could be moved outside the loop at line 207 because the variables it references won't change across iterations of the loop.